### PR TITLE
Add possibility to specify extra logs to dump

### PR DESCRIPTION
**Pull request by oshchirs**
Signed-off-by: Oleksandr Shchirskyi <oleksand.shchirskyi@intel.com>

### CODE REVIEW

**Review by katlapinka — DISMISSED** [review]

File: log/logger.py [hunk 1]
```diff
@@ -182,8 +182,10 @@ def get_additional_logs(self):
         if not check_if_file_exists(messages_log):
             messages_log = "/var/log/syslog"
         log_files = {"messages": messages_log,
-                     "dmesg": "/tmp/dmesg",
-                     "cas": "/var/log/opencas.log"}
+                     "dmesg": "/tmp/dmesg"}
+        if hasattr(TestRun.usr, "logs_to_dump"):
+            log_files.update(TestRun.usr.logs_to_dump)
```
Comment (robertbaldyga) [inline log/logger.py:187]: test-framework should never access `TestRun.usr`. This is private field reserved for TF user. Please use `dut_config` instead to pass list of paths. (You can see how plugin configuration is handled. The list of logs may be specified in `conftest.py` before `TestRun.prepare()` is called).

**Review by robertbaldyga — CHANGES_REQUESTED** [review]

File: log/logger.py (hunk 1 above)
Comment (oshchirs) [reply log/logger.py]: done

**Review by Deixx — APPROVED** [review]
I'm wondering if we should make a distinction between logs, which can be easily copied from a platform (i.e. `/var/log/messages`) and those that require dumping a command output first (i.e. `dmesg`, `cat /proc/mdstat`).
Sth like:
```
logs = { "messages": "/var/log/messages",
         "dmesg": "dmesg"
         "mdstat": "cat /proc/mdstat"
}

for log_name, log_source_path in log_files.items():
    if not file_exists(log_source_path):    # will not exist for "command logs"
        executor.run(f"{log_source_path} > /tmp/{log_name})
        log_source_path = "/tmp/" + log_name
    ...
    executor.rsync_from(log_source_path, log_destination_path)
```

**Review by katlapinka — APPROVED** [review]

**Review by robertbaldyga — APPROVED** [review]

**Comment by robertbaldyga** [comment]
> I'm wondering if we should make a distinction between logs, which can be easily copied from a platform (i.e. `/var/log/messages`) and those that require dumping a command output first (i.e. `dmesg`, `cat /proc/mdstat`).
> Sth like:
> 
> ```
> logs = { "messages": "/var/log/messages",
>          "dmesg": "dmesg"
>          "mdstat": "cat /proc/mdstat"
> }
> 
> for log_name, log_source_path in log_files.items():
>     if not file_exists(log_source_path):    # will not exist for "command logs"
>         executor.run(f"{log_source_path} > /tmp/{log_name})
>         log_source_path = "/tmp/" + log_name
>     ...
>     executor.rsync_from(log_source_path, log_destination_path)
> ```

This looks a bit fragile. We could make full-fledged API for that instead. Something like
```python
TestRun.attach_log(log_path, target_name=None, dump_fn=None)
```
where `dump_fn` would be a function that dumps the actual log to a file. That way we can even have logs that require more sophisticated extraction method, or logs that are filtered or preprocessed in some other way.

**Pull request merged**